### DISCUSSION TITLE
Text Detection with EAST and other methods

### DISCUSSION THREAD

**Discussion by Adhitya1996** [PaddlePaddle/PaddleOCR#13847] [Q&A]
Dear Paddle Community,

I have performed fine-tuning with en_PP-OCRv3_det with ch_PP-OCRv3_det_student.yml as config file. It has been attached below for your reference. 

Fine-tune configs file: https://github.com/PaddlePaddle/PaddleOCR/tree/main/configs/det/ch_PP-OCRv3

It is known that PaddleOCR has 4 text detection algorithms namely EAST, DB, SAST, PSENet. 

The default config file uses DB for text detection. I would like to train with EAST now to see if it performs better hence I looked into configs file: https://github.com/PaddlePaddle/PaddleOCR/tree/main/configs/det

![image](https://github.com/user-attachments/assets/08482d38-aa90-43af-9634-b6187439d9b0)

Issue: 
1. Config files that are present outside like det_mv3_db.yml, det_mv3_east.yml etc, seem to be used for full training of the text detection as the epoch number and other parameters suggests that.
Link for full training: https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/detection_en.md
![image](https://github.com/user-attachments/assets/3cdbf8b4-e7b5-4ee9-91a8-1fc8c508f96a)

Query: I humbly request for guidance on how to train text detection with EAST algorithm. Also, which weights to use for it?

2. I have converted EAST file into inference model with tools/export_model.py.

Still I am unable to run it like this:  ocr = PaddleOCR(use_gpu=True, det_algorithm='EAST',
                    det_model_dir='C:/Users/Adhi/paddle_fine_tuning/text_detection/05_det_EAST_inference',
                    use_angle_cls=True, lang='en')
Apparently only DB can be used as an argument.

I checked for alternative: https://github.com/PaddlePaddle/PaddleOCR/blob/main/doc/doc_en/inference_en.md

But I do not want to run like this: python3 tools/infer/predict_det.py --image_dir="./doc/imgs_en/img_10.jpg" --det_model_dir="./inference/det_east/" --det_algorithm="EAST"
 
rather in an integrated way like this: ocr = PaddleOCR(use_gpu=True, det_algorithm='EAST',
                    det_model_dir='C:/Users/Adhi/paddle_fine_tuning/text_detection/05_det_EAST_inference',
                    use_angle_cls=True, lang='en'), how to accomplish this?

Thanks in advance!

Best Regards,
Adhitya

**Comment by Adhitya1996** [comment]
Configs file:

Global:
  use_gpu: true
  epoch_num: 500
  log_smooth_window: 20
  print_batch_step: 10
  save_model_dir: C:/Users/Adhi/paddle_fine_tuning/text_detection/04_OCRv3_EAST_fine-tuned
  save_epoch_step: 100
  eval_batch_step:
  - 0
  - 400
  cal_metric_during_train: False
  pretrained_model: C:/Users/Adhi/paddle_fine_tuning/text_detection/03_en_PP-OCRv3_det_distill_train/student.pdparams
  checkpoints: null
  save_inference_dir: null
  use_visualdl: False
  infer_img: null
  save_res_path: C:/Users/Adhi/paddle_fine_tuning/text_detection/04_OCRv3_EAST_fine-tuned/predicts_east.txt

Architecture:
  model_type: det
  algorithm: EAST
  Transform:
  Backbone:
    name: MobileNetV3
    scale: 0.5
    model_name: large
  Neck:
    name: EASTFPN
    model_name: small
  Head:
    name: EASTHead
    model_name: small

Loss:
  name: EASTLoss
  
Optimizer:
  name: Adam
  beta1: 0.9
  beta2: 0.999
  lr:
    learning_rate: 0.001
  regularizer:
    name: 'L2'
    factor: 0

PostProcess:
  name: EASTPostProcess
  score_thresh: 0.8
  cover_thresh: 0.1
  nms_thresh: 0.2

Metric:
  name: DetMetric
  main_indicator: hmean

Train:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Adhi/paddle_fine_tuning/text_detection/01_det_images
    label_file_list: [C:/Users/Adhi/paddle_fine_tuning/text_detection/01_det_images/train/train_det.txt]
    ratio_list: [1.0]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - EASTProcessTrain:
          image_shape: [512, 512]
          background_ratio: 0.125
          min_crop_side_ratio: 0.1
          min_text_size: 10
      - KeepKeys:
          keep_keys: ['image', 'score_map', 'geo_map', 'training_mask'] # dataloader will return list in this order
  loader:
    shuffle: True
    drop_last: False
    batch_size_per_card: 1
    num_workers: 4

Eval:
  dataset:
    name: SimpleDataSet
    data_dir: C:/Users/Adhi/paddle_fine_tuning/text_detection/01_det_images
    label_file_list: [C:/Users/Adhi/paddle_fine_tuning/text_detection/01_det_images/test/test_det.txt]
    transforms:
      - DecodeImage: # load image
          img_mode: BGR
          channel_first: False
      - DetLabelEncode: # Class handling label
      - DetResizeForTest:
          limit_side_len: 2400
          limit_type: max
      - NormalizeImage:
          scale: 1./255.
          mean: [0.485, 0.456, 0.406]
          std: [0.229, 0.224, 0.225]
          order: 'hwc'
      - ToCHWImage:
      - KeepKeys:
          keep_keys: ['image', 'shape', 'polys', 'ignore_tags']
  loader:
    shuffle: False
    drop_last: False
    batch_size_per_card: 1 # must be 1
    num_workers: 2
